### ISSUE TITLE
slcli fails to work with "pkg_resources.DistributionNotFound: SoftLayer==5.0.1"

### ISSUE THREAD

**Issue by nimmoadam** [softlayer/softlayer-python#762]
### Environment Information
Operating System: Mac OS
softlayer-python version (`slcli --version`):

$ python --version
Python 2.7.12

$ pip -V
pip 8.1.2 from /usr/local/lib/python2.7/site-packages (python 2.7)

I was using slcli for more than 6 months now. Suddenly it stopped working with the following exception.

```
$ slcli
Traceback (most recent call last):
  File "/usr/local/bin/slcli", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 2793, in <module>
    working_set.require(__requires__)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 673, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 576, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: requests>=2.7.0
```

I didn't find much information in internet, but I did the following.. 

> sudo pip install requests==2.7.0 --force-reinstall
It was same again. So then,
I updated the requests,

> $ pip freeze | grep requests
> requests==2.11.1
> requests-toolbelt==0.6.2

That didn't work, so I reverted to 

> sudo pip install --upgrade --force-reinstall requests==2.7.0 tutum

Now I have

> $ pip freeze | grep requests
> requests==2.7.0
> requests-toolbelt==0.6.2

Then, I tried uninstalling slcli & installed it again.. Now, I am seeing pckg disctribution error for Softlayer version.
```
$ slcli
Traceback (most recent call last):
  File "/usr/local/bin/slcli", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 2793, in <module>
    working_set.require(__requires__)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 673, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/pkg_resources.py", line 576, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: SoftLayer==5.0.1
```

Can someone help in resolving this? Thanks!

**Comment by camporter** [comment]
What's the output of running `pip install --upgrade softlayer`?

**Comment by nimmoadam** [comment]
**pip install --upgrade softlayer**
```
Requirement already up-to-date: softlayer in /usr/local/lib/python2.7/site-packages
Requirement already up-to-date: prettytable>=0.7.0 in /Library/Python/2.7/site-packages/prettytable-0.7.2-py2.7.egg (from softlayer)
Collecting prompt-toolkit>=0.53 (from softlayer)
  Downloading prompt_toolkit-1.0.7-py2-none-any.whl (237kB)
    100% |████████████████████████████████| 245kB 1.4MB/s
Collecting six>=1.7.0 (from softlayer)
  Downloading six-1.10.0-py2.py3-none-any.whl
Requirement already up-to-date: click>=5 in /Library/Python/2.7/site-packages/click-6.6-py2.7.egg (from softlayer)
Collecting requests>=2.7.0 (from softlayer)
  Downloading requests-2.11.1-py2.py3-none-any.whl (514kB)
    100% |████████████████████████████████| 522kB 1.1MB/s
Requirement already up-to-date: pygments>=2.0.0 in /Library/Python/2.7/site-packages/Pygments-2.1.3-py2.7.egg (from softlayer)
Collecting wcwidth (from prompt-toolkit>=0.53->softlayer)
  Downloading wcwidth-0.1.7-py2.py3-none-any.whl
Installing collected packages: wcwidth, six, prompt-toolkit, requests
  Found existing installation: wcwidth 0.1.6
    Uninstalling wcwidth-0.1.6:
      Successfully uninstalled wcwidth-0.1.6
  Found existing installation: six 1.9.0
    Uninstalling six-1.9.0:
      Successfully uninstalled six-1.9.0
  Found existing installation: prompt-toolkit 0.60
    Uninstalling prompt-toolkit-0.60:
      Successfully uninstalled prompt-toolkit-0.60
  Found existing installation: requests 2.7.0
    Uninstalling requests-2.7.0:
      Successfully uninstalled requests-2.7.0
Successfully installed prompt-toolkit requests six wcwidth
Traceback (most recent call last):
  File "/usr/local/bin/pip", line 9, in <module>
    load_entry_point('pip==8.1.2', 'console_scripts', 'pip')()
  File "/usr/local/lib/python2.7/site-packages/pip/__init__.py", line 221, in main
    return command.main(cmd_args)
  File "/usr/local/lib/python2.7/site-packages/pip/basecommand.py", line 252, in main
    pip_version_check(session)
  File "/usr/local/lib/python2.7/site-packages/pip/utils/outdated.py", line 102, in pip_version_check
    installed_version = get_installed_version("pip")
  File "/usr/local/lib/python2.7/site-packages/pip/utils/__init__.py", line 848, in get_installed_version
    working_set = pkg_resources.WorkingSet()
  File "/usr/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 619, in __init__
    self.add_entry(entry)
  File "/usr/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 675, in add_entry
    for dist in find_distributions(entry, True):
  File "/usr/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1942, in find_eggs_in_zip
    if metadata.has_metadata('PKG-INFO'):
  File "/usr/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1463, in has_metadata
    return self.egg_info and self._has(self._fn(self.egg_info, name))
  File "/usr/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1824, in _has
    return zip_path in self.zipinfo or zip_path in self._index()
  File "/usr/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1704, in zipinfo
    return self._zip_manifests.load(self.loader.archive)
  File "/usr/local/lib/python2.7/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1644, in load
    mtime = os.stat(path).st_mtime
OSError: [Errno 2] No such file or directory: '/Library/Python/2.7/site-packages/prompt_toolkit-0.60-py2.7.egg'
```

**Comment by sudorandom** [comment]
It looks like there are two places where python modules are installed:

 * /usr/local/lib/python2.7/site-packages
 * /Library/Python/2.7/site-packages

Here's some more questions to help troubleshoot:
 * When you say "Then, I tried uninstalling slcli & installed it again" how did you go about uninstalling it?
 * Can you manually check the contents of both of the directories above for references of the SoftLayer package when you uninstall the SoftLayer package?
 * When the SoftLayer package is uninstalled, can you make sure that `/usr/local/bin/slcli` has been removed as well?
 * Can you give a directory listing for both of the directories listed above?
 * Can you run these two commands and give the output? `which pip` `sudo which pip`

I suspect that having two places where modules are installed is what is causing this behavior.

**Comment by sudorandom** [comment]
Ping @nimmoadam

**Comment by nimmoadam** [comment]
@sudorandom Thank you for the troubleshooting tip. Sorry missed notice your messages. 

- /usr/local/lib/python2.7/site-packages
- /Library/Python/2.7/site-packages

I checked in the above 2 locations. only in "/usr/local/lib/python2.7/site-packages" I see softlayer package. and I have this in that filepath
drwxr-xr-x  23 USER  admin   782B Aug 23 08:24 SoftLayer/
drwxr-xr-x   9 USER  admin   306B Aug 23 08:24 SoftLayer-5.1.0.dist-info/

```
 which pip
/usr/local/bin/pip

   sudo which pip
Password:
/usr/local/bin/pip
```

I don't remember how I uninstalled. I think, I have deleted that folder. 

`/usr/local/bin/slcli` is still present. 

Should I have to uninstall it by using some command & re-install it again? 
Attaching directory list of both the file_path you mentioned. 
library_python.txt --- contains files from (/Library/Python/2.7/site-packages)
usr_local.txt --- contains files from (/usr/local/lib/python2.7/site-packages)

Thank you!
